libs/core/langchain_core/output_parsers/json.py

```diff
@@ -35,7 +35,7 @@ def _custom_parser(multiline_string: str) -> str:
         multiline_string = multiline_string.decode()
 
     multiline_string = re.sub(
-        r'("action_input"\:\s*")(.*)(")',
+        r'("action_input"\:\s*")(.*?)(")',
```

Comment:
@nfcampos before this would just match everything before the last ", meaning it'd overwrite any fields after action_input